rts/Lua/LuaSyncedRead.cpp

```diff
@@ -3652,14 +3630,75 @@ int LuaSyncedRead::GetProjectilesInRectangle(lua_State* L)
 			if (!LuaUtils::IsProjectileVisible(L, pro))
 				continue;
 
-			lua_pushnumber(L, pro->id);
-			lua_rawseti(L, -2, arrayIndex++);
+			lua_pushnumber(L, static_cast<lua_Number>(pro->id));
+			lua_rawseti(L, -2, static_cast<int>(arrayIndex++));
 		}
 	}
+}
+
+/***
+ *
+ * @function Spring.GetProjectilesInRectangle
+ * @param xmin number
+ * @param zmin number
+ * @param xmax number
+ * @param zmax number
+ * @param excludeWeaponProjectiles boolean? (Default: false)
+ * @param excludePieceProjectiles boolean? (Default: false)
+ * @return number[] projectileIDs
+ */
+int LuaSyncedRead::GetProjectilesInRectangle(lua_State* L)
+{
+	const float xmin = luaL_checkfloat(L, 1);
+	const float zmin = luaL_checkfloat(L, 2);
+	const float xmax = luaL_checkfloat(L, 3);
+	const float zmax = luaL_checkfloat(L, 4);
+
+	const bool excludeWeaponProjectiles = luaL_optboolean(L, 5, false);
+	const bool excludePieceProjectiles = luaL_optboolean(L, 6, false);
 
+	const float3 mins(xmin, 0.0f, zmin);
+	const float3 maxs(xmax, 0.0f, zmax);
+
+	QuadFieldQuery qfQuery;
+	quadField.GetProjectilesExact(qfQuery, mins, maxs);
+	GetProjectilesLuaTable(L, qfQuery, excludeWeaponProjectiles, excludePieceProjectiles);
 	return 1;
 }
 
+/***
+ *
+ * @function Spring.GetProjectilesInSphere
+ * @param x number
+ * @param y number
+ * @param z number
+ * @param radius number
+ * @param excludeWeaponProjectiles boolean? (Default: false)
+ * @param excludePieceProjectiles boolean? (Default: false)
+ * @return number[] projectileIDs
+ */
+int LuaSyncedRead::GetProjectilesInSphere(lua_State* L)
+{
+	const float x = luaL_checkfloat(L, 1);
+	// needed for consistency with the rest of the spherical API
+	[[maybe_unused]] const float y = luaL_checkfloat(L, 2);
+	const float z = luaL_checkfloat(L, 3);
+	const float circleRadius = luaL_checkfloat(L, 4);
+
+	const bool excludeWeaponProjectiles = luaL_optboolean(L, 5, false);
+	const bool excludePieceProjectiles = luaL_optboolean(L, 6, false);
+
+	constexpr auto hexagonApothem = static_cast<float>(std::numbers::sqrt3 / 2);
+	const float scaledRadius = circleRadius * hexagonApothem;
+
+	float3 mins(x - scaledRadius, 0.0f, z - scaledRadius);
+	float3 maxs(x + scaledRadius, 0.0f, z + scaledRadius);
```

Comment:
Looks like this actually gets projectiles in a square inscribed into a circle rather than a sphere?